Mobile-friendly Bootstrap Tables

beekman opened the pull request:
### Any helpful knowledge/context for the reviewer?
npm install is required, for the useMediaQuery hook.

### Please make sure you've attempted to meet the following coding standards
- [X] Code builds successfully
- [X] Code has been tested and does not produce errors
- [X] Code is readable and formatted
- [X] There isn't any unnecessary commented-out code
- [X] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

beekman committed:
mobile properties table with media query hook

beekman committed:
responsive tables implemented on managers page

beekman commented:
@Chris-Boe, the hook-based solution we came up with is working great for the pages with functional components. However, the tickets page is a class-based component. I'm not sure what the best solution is to adapt the react hook we are using for breakpoint matching on the other pages. Do you have any suggestions how to proceed?

Chris-Boe commented:
> @Chris-Boe, the hook-based solution we came up with is working great for the pages with functional components. However, the tickets page is a class-based component. I'm not sure what the best solution is to adapt the react hook we are using for breakpoint matching on the other pages. Do you have any suggestions how to proceed?

If you want you could take a stab at converting the class component to a functional component! The other pages with grids should be good examples for what the end goal would be.
If you don't want to get into that yet, I would probably leave the Tickets table alone and we can do it in a separate issue but get these other tables updated in the meantime

beekman committed:
correcting version number for media query in package.json

beekman committed:
tenants Bootstrap Table now responsive except expandRow definition, and fix for properties page

beekman committed:
tenants page expandRow definition and row widths for both mobile and desktop

beekman committed:
fix for expandable row details for properties

beekman committed:
completed row expansion on mobile for all React tables except the tickets component

beekman committed:
refactoring fix for tenants component

beekman committed:
additional refactoring of tenants component

beekman committed:
tenants expandRow fixes

beekman committed:
fixes for multiple invalid warnings for links without valid href values, mostly implemented breakpoint code for tickets page

beekman committed:
finished mobile react table breakpoint implementation for tickets page

Chris-Boe commented on src/views/managers/managers.jsx:
```diff
@@ -123,13 +60,61 @@ const getManagers = (header, storeInState, updateLoading) => {
     });
 };
 
+const expandRow = isSmallScreen => ({
```
We could probably move this out to the managersFormStructure page as well and import it in this file

beekman replied:
Unfortunately, we can't, as it uses a hook as a parameter, and hooks have to be called from their components.

Chris-Boe commented on src/views/properties/properties.jsx:
```diff
@@ -6,59 +6,58 @@ import * as axios from 'axios';
-const columns = [{
-  dataField: 'name',
-  formatter: (cell, row, rowIndex, formatExtraData) => {
+const expandRow = isSmallScreen => ({
```
I think we could also move this one out into propertiesFormStructure

beekman replied:
Unfortunately, we can't, as it uses a hook as a parameter, and hooks have to be called from their components.

Chris-Boe commented on src/views/properties/properties.jsx:
```diff
@@ -6,59 +6,58 @@ import * as axios from 'axios';
+
+const makeAuthHeaders = ({ user }) => ({ headers: { 'Authorization': `Bearer ${user.accessJwt}` } });
+
+
```
There's quite a few empty lines here that we could remove!

beekman replied:
I've just pushed a commit that removes the excessive line breaks.

Chris-Boe commented on src/views/properties/propertiesFormStructure.jsx:
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
```
I think we could rename all these FormStructure files to something like TableComponents since there aren't forms on the pages

beekman replied:
I just renamed all of the three that we are able to use! Unfortunately the tickets page we can't do this for, because the definitions on that page require additional parameters for the unique link wrapper being used on that page..

Chris-Boe commented on src/views/tenants/tenants.jsx:
```diff
@@ -49,6 +52,33 @@ const getDisplayTenants = (tenants, showHoused, showArchived) => {
   return tenants.filter(tenant => (showHoused || tenant.lease) && (showArchived || !tenant.archived));
 };
 
+const expandRow = isSmallScreen => ({
```
Same here

beekman replied:
Unfortunately, we can't, as it uses a hook as a parameter, and hooks have to be called from their components.

Chris-Boe commented on src/views/tickets/tickets.jsx:
```diff
@@ -62,19 +63,62 @@ const options = {
   pageButtonRenderer
 };
 
+const expandRow = isSmallScreen => ({
```
We can pull out this, the mobileColumns, and the pageButtonRenderer to the components file

beekman replied:
Unfortunately, we can't, as it uses a hook as a parameter, and hooks have to be called from their components.

Chris-Boe reviewed (changes requested):
This looks great! There's a couple organizational improvements I think we could make but the mobile tables are looking nice

beekman commented:
Great feedback! Originally I was planning to move the expandRow definitions
but ran into a complication. The problem I was running into  is that those
definitions use the breakpoint hook as a parameter, and hooks need to be
called from the components themselves due to the rules of hooks. so unless
you have a workaround for that we might be unable to do that.

On Mon, Jul 5, 2021 at 9:56 PM Chris Boe ***@***.***> wrote:

> *@Chris-Boe* requested changes on this pull request.
>
> This looks great! There's a couple organizational improvements I think we
> could make but the mobile tables are looking nice
> ------------------------------
>
> In src/views/managers/managers.jsx
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664216586>
> :
>
> > @@ -123,13 +60,61 @@ const getManagers = (header, storeInState, updateLoading) => {
>      });
>  };
>
> +const expandRow = isSmallScreen => ({
>
> We could probably move this out to the managersFormStructure page as well
> and import it in this file
> ------------------------------
>
> In src/views/properties/properties.jsx
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664217139>
> :
>
> >
> -const columns = [{
> -  dataField: 'name',
> -  formatter: (cell, row, rowIndex, formatExtraData) => {
> +const expandRow = isSmallScreen => ({
>
> I think we could also move this one out into propertiesFormStructure
> ------------------------------
>
> In src/views/properties/properties.jsx
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664217244>
> :
>
> > -  headerStyle: () => {
> -    return { width: "10%" };
> -  }
> -}, {
> -  dataField: 'created_at',
> -  text: 'Added On',
> -  sort: true,
> -  headerStyle: () => {
> -    return { width: "10%" };
> -  }
> -}];
> +});
> +
> +const makeAuthHeaders = ({ user }) => ({ headers: { 'Authorization': `Bearer ${user.accessJwt}` } });
> +
> +
>
> There's quite a few empty lines here that we could remove!
> ------------------------------
>
> In src/views/properties/propertiesFormStructure.jsx
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664229130>
> :
>
> > @@ -0,0 +1,75 @@
> +import React from 'react';
>
> I think we could rename all these FormStructure files to something like
> TableComponents since there aren't forms on the pages
> ------------------------------
>
> In src/views/tenants/tenants.jsx
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664230159>
> :
>
> > @@ -49,6 +52,33 @@ const getDisplayTenants = (tenants, showHoused, showArchived) => {
>    return tenants.filter(tenant => (showHoused || tenant.lease) && (showArchived || !tenant.archived));
>  };
>
> +const expandRow = isSmallScreen => ({
>
> Same here
> ------------------------------
>
> In src/views/tickets/tickets.jsx
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664230641>
> :
>
> > @@ -62,19 +63,62 @@ const options = {
>    pageButtonRenderer
>  };
>
> +const expandRow = isSmallScreen => ({
>
> We can pull out this, the mobileColumns, and the pageButtonRenderer to the
> components file
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/codeforpdx/dwellingly-app/pull/590#pullrequestreview-699513296>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEJQIMJNHOXUR7OVR4TUGTTWKEHLANCNFSM43PMQ3MA>
> .
>
-- 
-- 
Ben Beekman
h <http://www.beekmedia.com>ttps://benbeekman.dev

Chris-Boe commented:
> Great feedback! Originally I was planning to move the expandRow definitions but ran into a complication. The problem I was running into is that those definitions use the breakpoint hook as a parameter, and hooks need to be called from the components themselves due to the rules of hooks. so unless you have a workaround for that we might be unable to do that.
> On Mon, Jul 5, 2021 at 9:56 PM Chris Boe ***@***.***> wrote: *@Chris-Boe* requested changes on this pull request. This looks great! There's a couple organizational improvements I think we could make but the mobile tables are looking nice ------------------------------ In src/views/managers/managers.jsx <[#590 (comment)](https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664216586)> : > @@ -123,13 +60,61 @@ const getManagers = (header, storeInState, updateLoading) => { }); }; +const expandRow = isSmallScreen => ({ We could probably move this out to the managersFormStructure page as well and import it in this file ------------------------------ In src/views/properties/properties.jsx <[#590 (comment)](https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664217139)> : > -const columns = [{ - dataField: 'name', - formatter: (cell, row, rowIndex, formatExtraData) => { +const expandRow = isSmallScreen => ({ I think we could also move this one out into propertiesFormStructure ------------------------------ In src/views/properties/properties.jsx <[#590 (comment)](https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664217244)> : > - headerStyle: () => { - return { width: "10%" }; - } -}, { - dataField: 'created_at', - text: 'Added On', - sort: true, - headerStyle: () => { - return { width: "10%" }; - } -}]; +}); + +const makeAuthHeaders = ({ user }) => ({ headers: { 'Authorization': `Bearer ${user.accessJwt}` } }); + + There's quite a few empty lines here that we could remove! ------------------------------ In src/views/properties/propertiesFormStructure.jsx <[#590 (comment)](https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664229130)> : > @@ -0,0 +1,75 @@ +import React from 'react'; I think we could rename all these FormStructure files to something like TableComponents since there aren't forms on the pages ------------------------------ In src/views/tenants/tenants.jsx <[#590 (comment)](https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664230159)> : > @@ -49,6 +52,33 @@ const getDisplayTenants = (tenants, showHoused, showArchived) => { return tenants.filter(tenant => (showHoused || tenant.lease) && (showArchived || !tenant.archived)); }; +const expandRow = isSmallScreen => ({ Same here ------------------------------ In src/views/tickets/tickets.jsx <[#590 (comment)](https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r664230641)> : > @@ -62,19 +63,62 @@ const options = { pageButtonRenderer }; +const expandRow = isSmallScreen => ({ We can pull out this, the mobileColumns, and the pageButtonRenderer to the components file — You are receiving this because you were assigned. Reply to this email directly, view it on GitHub <[#590 (review)](https://github.com/codeforpdx/dwellingly-app/pull/590#pullrequestreview-699513296)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAEJQIMJNHOXUR7OVR4TUGTTWKEHLANCNFSM43PMQ3MA> .
> -- -- Ben Beekman h <http://www.beekmedia.com>ttps://benbeekman.dev

Ah okay! Well we can leave the expandRow objects in those files then and look at it later if we still feel like it. Thanks for checking!

beekman committed:
removing excessive line breaks

beekman reviewed (commented):
Thanks for the review!

beekman committed:
renaming all react table components

beekman reviewed (commented):
I think this addresses the last of your feedback! Ready to merge if you agree!

Chris-Boe commented on src/views/tickets/tickets.jsx:
```diff
@@ -120,85 +120,6 @@ export function Tickets(props) {
     setViewedTicket((prevState) => (prevState ? null : ticket));
   };
 
-  const columns = [{
```
Looks like these and the ticketFormStructure got removed by accident!

Chris-Boe replied:
Hey @beekman ! I went ahead and replaced the missing components and fixed up a couple things so we can get this merged. I hope that's alright with you and you're doing alright!

Chris-Boe committed:
Merge branch 'development' into issue-548-mobile-tables

beekman commented:
Thanks so much @Chris-Boe, I was about to make the fix and glad to see you
beat me to it.

On Wed, Aug 4, 2021 at 8:08 PM Chris Boe ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/views/tickets/tickets.jsx
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r683095026>
> :
>
> > @@ -120,85 +120,6 @@ export function Tickets(props) {
>      setViewedTicket((prevState) => (prevState ? null : ticket));
>    };
>
> -  const columns = [{
>
> Hey @beekman <https://github.com/beekman> ! I went ahead and replaced the
> missing components and fixed up a couple things so we can get this merged.
> I hope that's alright with you and you're doing alright!
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/codeforpdx/dwellingly-app/pull/590#discussion_r683095026>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEJQIOQYGE2JXKLT5UREBDT3H6BZANCNFSM43PMQ3MA>
> .
>
-- 
-- 
Ben Beekman
h <http://www.beekmedia.com>ttps://benbeekman.dev
